src/MDMTransaction.m

```diff
@@ -30,17 +32,58 @@ - (instancetype)init {
 }
 
 - (void)addPlan:(NSObject<MDMPlan> *)plan toTarget:(id)target {
-  MDMTransactionLog *log = [MDMTransactionLog new];
-  log.plans = @[ [plan copy] ];
-  log.target = target;
+  [self commonAddPlan:plan toTarget:target withName:nil];
+}
+
+- (void)addPlan:(NSObject<MDMPlan> *)plan toTarget:(id)target named:(NSString *)name {
+  if (name.length && [self planForName:name withTarget:target]) {
```

Comment:
I see you're special-casing the empty string here; can I learn more about your rationale here?

If we do decide to go this route we'll need to document the behavior, which leads me to wonder if we should be special-casing the behavior or simply allowing empty strings as names.